src/detect_breakends/junction_detection.cpp

```diff
@@ -102,19 +100,7 @@ void detect_junctions_in_alignment_file(const std::filesystem::path & alignment_
                                       insertion_file);
                         break;
                     case 2: // Detect junctions from split read evidence (SA tag, primary alignments only)
-                        if (!hasFlagSupplementary(flag))
-                        {
-                            std::string sa_tag = tags.get<"SA"_tag>();
-                            if (!sa_tag.empty())
-                            {
-                                std::vector<AlignedSegment> aligned_segments{};
-                                auto strand = (hasFlagReverseComplement(flag) ? strand::reverse : strand::forward);
-                                aligned_segments.push_back(AlignedSegment{strand, ref_name, pos, mapq, cigar});
-                                retrieve_aligned_segments(sa_tag, aligned_segments);
-                                std::sort(aligned_segments.begin(), aligned_segments.end());
-                                analyze_aligned_segments(aligned_segments, junctions, query_name);
-                            }
-                        }
+                        analyze_sa_tag(query_name, flag, ref_name, pos, mapq, cigar, tags, junctions);
```

Comment:
I would leave the check of the supplementary flag and the check whether the SA tag is empty out of the method:

```
                    case 2: // Detect junctions from split read evidence (SA tag, primary alignments only)
                        if (!hasFlagSupplementary(flag))
                        {
                            std::string sa_tag = tags.get<"SA"_tag>();
                            if (!sa_tag.empty())
                            {
                                analyze_sa_tag(query_name, flag, ref_name, pos, mapq, cigar, tags, junctions);
                            }
                        }
                        break;
```

This makes it clearer what is actually happening without looking into the method. To me, that feels much more natural and makes sure that the method carries out only the well-defined task of analyzing the SA tag.